azure-iot-device/azure/iot/device/connection_string.py

```diff
@@ -93,18 +91,25 @@ def _parse_connection_string(connection_string):
 def _validate_keys(d):
     """Raise ValueError if incorrect combination of keys in dict d"""
     host_name = d.get(HOST_NAME)
-    shared_access_key_name = d.get(SHARED_ACCESS_KEY_NAME)
     shared_access_key = d.get(SHARED_ACCESS_KEY)
+    shared_access_signature = d.get(SHARED_ACCESS_SIGNATURE)
     device_id = d.get(DEVICE_ID)
     x509 = d.get(X509)
 
-    if shared_access_key and x509 and x509.lower() == "true":
+    # Validate only one type of auth included
+    auth_count = 0
+    if shared_access_key:
+        auth_count += 1
+    if x509 and x509.lower() == "true":
```

Comment:
Do we have to worry about x509 being truthy but not "true"

Reply:
It shouldn't happen, but it's possible by definition of the boolean keyword.